numpy/polynomial/legendre.py

```diff
@@ -1556,6 +1576,18 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
     c, resids, rank, s = la.lstsq(lhs.T/scl, rhs.T, rcond)
     c = (c.T/scl).T
 
+    # Expand c to include non-fitted coefficients, which are set to
+    # zero
+    if restricted_L:
```

Comment:
Interestingly, the original version (commit 21fb38f) did exactly as you suggested i.e. starting with an array of zeros. Easy enough to drop the patch that moves to the insert approach.